src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java

```diff
@@ -966,11 +968,30 @@ private Map<GameboardDTO, List<String>> extractGameboardQuestionIds(
                         questionIds = Lists.newArrayList();
                     }
                     questionIds.add(question.getId());
+                    questionIdToPageIdMap.put(question.getId(), questionPage.getId());
                     gameboardQuestionIds.put(gameboard, questionIds);
                 }
             }
         }
-        return gameboardQuestionIds;
+        return new QuestionIdMaps(gameboardQuestionIds, questionIdToPageIdMap);
+    }
+
+    private static class QuestionIdMaps {
+        private final Map<GameboardDTO, List<String>> gameboardToQuestionMap;
```

Comment:
Small one, but shouldn't this be "questions", in plural? Since it's a list of strings.